### PR TITLE
{chem}[foss/2022a] LAMMPS v23Jun2022, ScaFaCoS v1.0.4, Voro++ v0.4.6, kim-api v2.3.0, archspec v0.1.4

### PR DESCRIPTION
**Pull request by maxim-masterov**
Add LAMMPS v23Jun2022 and dependencies for the foss/2022a toolchain

Depends on PR #16007

### CODE REVIEW

**Comment by casparvl** [comment]
In the conf-call of https://github.com/easybuilders/easybuild/wiki/Conference-call-notes-20220928 we discussed the FFmpeg version to use in 2022a. The decision was to stick with 4.4.2. Main reasons are to maintain compatibility with the version used in Paraview (which is not yet compatible with 5.X) and to avoid running into other software in the future that would need 4.4.2 (which would create more incompatible modules in the 2022a toolchain). Once 5.X is matured a bit, we can make the jump there in a subsequent toolchain.

@maxim-masterov  Could you change the FFmpeg version used in this PR?

**Comment by maxim-masterov** [comment]
@casparvl done

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2022a-kokkos.eb [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    'SHOCK',
+    'SMTBQ',
+    'SPH',
+    'SPIN',
+    'SRD',
+    'TALLY',
+    'UEF',
+    'VORONOI',
+    'VTK',
+    'YAFF',
```
Comment (ocaisa) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2022a-kokkos.eb:147]: Just wondering if you've checked this list against https://docs.lammps.org/Packages_details.html and https://docs.lammps.org/Build_extras.html to see if there's any other low-hanging fruit we can include?

Comment (maxim-masterov) [reply easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2022a-kokkos.eb:155]: thanks for spotting it, I've added/removed some packages from the list

Comment (ocaisa) [reply easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2022a-kokkos.eb:155]: I think I've hurt more than I've helped here! Some of the now included packages (kokkos, GPU, intel) are already handled by the easyblock

Comment (maxim-masterov) [reply easybuild/easyconfigs/l/LAMMPS/LAMMPS-23Jun2022-foss-2022a-kokkos.eb:152]: Missed it... I've also removed OPENMP as it is handled by the easyblock too.

**Comment by ocaisa** [comment]
@maxim-masterov They just released [Update 4 for `23Jun2022`](https://github.com/lammps/lammps/releases/tag/stable_23Jun2022_update4), since this is not merged yet, would be good to use those sources

**Comment by maxim-masterov** [comment]
@ocaisa updated LAMMPS version, and PLUMED and VTK dependencies

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
zen2-ubuntu-2304 - Linux Ubuntu 23.04 (Lunar Lobster), x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.11.2
See https://gist.github.com/SebastianAchilles/8fb02d30dd3751e6c773a94eb1c2eb2a for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 525.105.17, Python 3.9.16
See https://gist.github.com/SebastianAchilles/90c8291a21b5b7989589f7955d28cb48 for a full test report.

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 5 out of 5 (4 easyconfigs in total)
tcn37.local.snellius.surf.nl - Linux Rocky Linux 8.7, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/dd65fbcece6961dd6264b9a3631e5377 for a full test report.

**Review by ocaisa — APPROVED** [review]
Lots of successful builds here, going in!

**Pull request merged**